ckanext/spatial/tests/test_harvest.py

```diff
@@ -0,0 +1,1165 @@
+from ckanext.spatial.tests.base import SpatialTestBase
+
+from ckanext.spatial.tests.xml_file_server import serve
+
+# Start simple HTTP server that serves XML test files
+serve()
+
+
+@pytest.mark.usefixtures('clean_db', 'clean_index', 'harvest_setup', 'spatial_setup')
+class HarvestFixtureBase(SpatialTestBase):
+
+    def setup(self):
+        # Add sysadmin user
+        harvest_user = model.User(name=u'harvest', password=u'test', sysadmin=True)
+        Session.add(harvest_user)
+        Session.commit()
+
+        package_schema = default_update_package_schema()
+        self.context ={'model':model,
+                       'session':Session,
+                       'user':u'harvest',
+                       'schema':package_schema,
+                       'api_version': '2'}
+
+    def teardown(self):
+       model.repo.rebuild_db()
+
+    def _create_job(self,source_id):
+        # Create a job
+        context ={'model':model,
+                 'session':Session,
+                 'user':u'harvest'}
+
+        job_dict=get_action('harvest_job_create')(context,{'source_id':source_id})
+        job = HarvestJob.get(job_dict['id'])
+        assert job
+
+        return job
+
+    def _create_source_and_job(self, source_fixture):
+        context ={'model':model,
+                 'session':Session,
+                 'user':u'harvest'}
+
+        if config.get('ckan.harvest.auth.profile') == u'publisher' \
+           and not 'publisher_id' in source_fixture:
+           source_fixture['publisher_id'] = self.publisher.id
+
+        source_dict=get_action('harvest_source_create')(context,source_fixture)
+        source = HarvestSource.get(source_dict['id'])
+        assert source
+
+        job = self._create_job(source.id)
+
+        return source, job
+
+    def _run_job_for_single_document(self,job,force_import=False,expect_gather_errors=False,expect_obj_errors=False):
+
+        harvester = GeminiDocHarvester()
+
+        harvester.force_import = force_import
+
+        object_ids = harvester.gather_stage(job)
+        assert object_ids, len(object_ids) == 1
+        if expect_gather_errors:
+            assert len(job.gather_errors) > 0
+        else:
+            assert len(job.gather_errors) == 0
+
+        assert harvester.fetch_stage(object_ids) == True
+
+        obj = HarvestObject.get(object_ids[0])
+        assert obj, obj.content
+
+        harvester.import_stage(obj)
+        Session.refresh(obj)
+        if expect_obj_errors:
+            assert len(obj.errors) > 0
+        else:
+            assert len(obj.errors) == 0
+
+        job.status = u'Finished'
+        job.save()
+
+        return obj
+
+class TestHarvest(HarvestFixtureBase):
+
+    @classmethod
+    def setup_class(cls):
+        SpatialHarvester._validator = Validators(profiles=['gemini2'])
+        # HarvestFixtureBase.setup_class()
+
+    def clean_tags(self, tags):
+        return list(map(lambda x: {u'name': x['name']}, tags))
```

Comment:
```suggestion
        return [{u'name': tag['name']} for tag in tags]
```
seems clearer in this case